cmd/virt-tail/main.go

```diff
@@ -26,169 +26,77 @@ import (
 	"fmt"
 	"os"
 	"os/signal"
-	"path/filepath"
 	"strconv"
-	"strings"
+	"syscall"
 	"time"
 
-	"github.com/fsnotify/fsnotify"
 	"github.com/nxadm/tail"
 	"github.com/spf13/pflag"
 	"golang.org/x/sync/errgroup"
-	"k8s.io/apimachinery/pkg/util/wait"
 
 	"kubevirt.io/client-go/log"
-
-	virtwait "kubevirt.io/kubevirt/pkg/apimachinery/wait"
 )
 
-// initial timeout for serial console socket creation
-const initialSocketTimeout = time.Second * 20
-
-type TermFileError struct{}
-type SocketFileError struct{}
-
-func (m *TermFileError) Error() string {
-	return "termFile got detected"
-}
-
-func (m *SocketFileError) Error() string {
-	return "socketFile got removed"
-}
-
 type VirtTail struct {
-	ctx           context.Context
-	logFile       string
-	g             *errgroup.Group
-	socketTimeout *time.Duration
+	ctx     context.Context
+	logFile string
 }
 
-func (v *VirtTail) checkFile(socketFile string) bool {
-	_, err := os.Stat(socketFile)
-	return !os.IsNotExist(err)
+func (v *VirtTail) tailLogsWrapper() error {
+	location := &tail.SeekInfo{Offset: 0}
+	for {
+		var err error
+		location, err = v.tailLogs(*location)
+		if err != nil {
+			return err
+		}
+		if location == nil {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
+	}
 }
 
-func (v *VirtTail) tailLogs() error {
+func (v *VirtTail) tailLogs(location tail.SeekInfo) (*tail.SeekInfo, error) {
 	t, err := tail.TailFile(v.logFile, tail.Config{
+		Location:      &location,
 		Follow:        true,
 		CompleteLines: true,
 		MustExist:     false,
 		ReOpen:        true,
 		Logger:        tail.DiscardingLogger,
 	})
 	if err != nil {
-		return err
+		return nil, err
 	}
+	cleanup := true
 	defer func() {
 		serr := t.Stop()
 		if serr != nil {
 			log.Log.V(3).Infof("tail error: %v", serr)
 		}
-		t.Cleanup()
+		if cleanup {
+			t.Cleanup()
+		}
 	}()
 
 	for {
 		select {
 		case line, ok := <-t.Lines:
 			if !ok {
-				log.Log.V(4).Info("tail error: line not ok")
+				log.Log.V(4).Info("tail error: chan closed")
+				cleanup = false
```

Comment:
I'm not sure I fully understand this logic. Could you please explain why cleanup is not needed? 
I think a comment is needed.